### DISCUSSION TITLE
Link between documents

### DISCUSSION THREAD

**Discussion by Matthias84** [jonaswinkler/paperless-ng#564] [Feature Requests]
Hi, I like to ask, if a feature might be useful to reference / group documents directly? Currently you can make use of tags to group documents, but IMHO this isn't useful if you like to refer only 2..3 documents regularly. Think about a empty form document and the same form but filled, or similar usecases.

**Comment by shamoon** [comment]
You mean like a "related documents" field?

**Comment by anokfireball** [comment]
For large document collections that are (possibly) shared by multiple people, this sounds like a good idea. [Firefly III](https://github.com/firefly-iii/firefly-iii) (selfhosted personal finance manager) has a similar option to logically connect multiple transactions:
![Screenshot from 2021-01-24 20-03-49](https://user-images.githubusercontent.com/10155170/105640707-a65e7d80-5e7f-11eb-929d-e80b7698797d.png)
Having pre-defined options works well in the case of "money goes from A to B". In the context of a DMS, it is less obvious what defaults should be, just because document types are arbitrary. Just a few that quickly come to mind:

- confirms
- cancels
- follows / preceeds
- superseeds
- extends / is part of

**Comment by Matthias84** [comment]
> You mean like a "related documents" field?

Similar, but based on a explicit user action, as @C0nsultant  points out. Think of it like referencing tickets in a bugtracker, where they can be linked, but also grouped via milestones?

**Comment by jonaswinkler** [comment]
Not sure about how to properly do a good UI for selecting target documents. I can see the appeal of links, especially for the mail consumer (link multiple documents from one mail automatically).

However, I'm also pretty close to declaring this as out of scope.

**Comment by d8sychain** [comment]
Perhaps when the custom metadata feature #274 is done it could be used in a way to relate documents.
Doc A, Doc B, Doc C get a custom field "Reference Number: 123"
Doc D, Doc E get a custom field "Reference Number: 456"

Searching for "Reference Number: 456" would result in the 2 related document

To expand on this, setting a custom field to a specific type, say "ref-#" (e.g. long-text, number, date, ref-#) causes the UI to list other documents on Doc D that match the set "ref-#". Some users might label the custom field e.g. Job Number, P.O. Number, Reference Number, but functions the same based on the field type.

**Comment by fdw** [comment]
Another, much simpler idea would be to list other documents from the same correspondent and for the same tags.

**Comment by seijikun** [comment]
Some of the similar tools I looked at solve such use-cases by splitting the architecture into documents and files.
A document can contain 1:n files. Like this, they basically support document groups and "linking".
But this may not map well to how paperless was designed.
